src/components/SchemaRow.tsx

```diff
@@ -1,128 +1,116 @@
 import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
-import { isReferenceNode, isRegularNode, ReferenceNode, SchemaNode, SchemaNodeKind } from '@stoplight/json-schema-tree';
-import { Box, Flex, Icon } from '@stoplight/mosaic';
-import { IRowRendererOptions, isParentNode, Tree } from '@stoplight/tree-list';
-import { Optional } from '@stoplight/types';
+import {
+  isMirroredNode,
+  isReferenceNode,
+  isRegularNode,
+  ReferenceNode,
+  SchemaNode,
+  SchemaNodeKind,
+} from '@stoplight/json-schema-tree';
+import { Box, Flex, Icon, VStack } from '@stoplight/mosaic';
 import * as React from 'react';
 
 import { CARET_ICON_BOX_DIMENSION, CARET_ICON_SIZE, SCHEMA_ROW_OFFSET } from '../consts';
-import { SchemaNodeContext, TreeListNodeContext } from '../contexts';
+import { useJSVOptionsContext } from '../contexts';
 import { isCombiner } from '../guards/isCombiner';
-import { useSchemaNode, useSchemaTree, useTreeListNode } from '../hooks';
-import { GoToRefHandler, SchemaTreeListNode } from '../types';
-import { isPropertyRequired } from '../utils/isPropertyRequired';
-import { Caret, Description, Divider, getValidationsFromSchema, Property, Validations } from './shared';
-import { Format } from './shared/Format';
+import { calculateChildrenToShow, isFlattenableNode, isPropertyRequired } from '../tree';
+import { Caret, Description, Divider, Format, getValidationsFromSchema, Property, Validations } from './shared';
 import { Properties } from './shared/Properties';
 
-export interface ISchemaRow {
-  className?: string;
-  treeListNode: SchemaTreeListNode;
-  rowOptions: IRowRendererOptions;
-  onGoToRef?: GoToRefHandler;
+export interface SchemaRowProps {
+  schemaNode: SchemaNode;
+  nestingLevel: number;
```

Comment:
I could perhaps make it a part of `SchemaNode`.
This is likely to be useful overall.

Reply:
Ah, actually, there's a property called `depth`.
https://github.com/stoplightio/json-schema-tree/blob/beta/src/nodes/BaseNode.ts#L19
Did you run into issues when trying to use this?

Reply:
The problem with that is exactly what you suggest. There is no 1-1 relationship between JST depth, and JSV depth because of array flattening. I don't think this prop adds much noise though, I think it's much more straightforward logic than trying to deduce the real depth from `schemaNode.depth` and its ancestor tree.